### PR TITLE
A Tiny JS World

### PR DESCRIPTION
**Pull request by OlStani**
###A Tiny JS World

[Demo](https://olstani.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/OlStani/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/OlStaniev/tiny-js/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const man = {
+   name: 'Mario',
+   gender: 'male',
+   legs: 2,
+   hands: 2,
+   saying: 'Hi!'
+}
+
+const inhabitants = [cat, dog, woman, man]
+const inhabitantsInformation = inhabitants.map(({name, gender, legs, hands,saying}) => [`${name}; ${gender}; ${legs}; ${hands}; ${saying}`])
```
Comment (OleksiyRudenko) [inline submissions/OlStaniev/tiny-js/index.js:34]: Having props explicitly listed (`{name, gender, legs, hands,saying}`) is a good approach.
Please use `#Array.map` and `#Array.join` to build the presentation string in the callback function. Remember, you can pack arguments into an array, e.g. `({a1, a2}) => [a1, a2]......some array processing code....`

Also why do you need to wrap every presentation string into an array? You have a nested arrays as a 

On your demo page, open dev console and type `inhabitantsInformation` to see the data structure.

Feel free seeking help in students' chat.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
Good job!
A couple of fixes and we are done.

File: submissions/OlStaniev/tiny-js/index.js (hunk 1 above)
Comment (OlStani) [reply submissions/OlStaniev/tiny-js/index.js:34]: > Please use `#Array.map` and `#Array.join` to build the presentation string in the callback function. Remember, you can pack arguments into an array, e.g. `({a1, a2}) => [a1, a2]......some array processing code....`

я не зовсім розумію що я маю тут зробити : Please use #Array.map and #Array.join to build the presentation string in the callback function. Remember, you can pack arguments into an array, e.g. ({a1, a2}) => [a1, a2]......some array processing code....

Comment (OleksiyRudenko) [reply submissions/OlStaniev/tiny-js/index.js:34]: Let's start with answering questions.
> Also why do you need to wrap every presentation string into an array? You have nested arrays as a result.

Comment (OlStani) [reply submissions/OlStaniev/tiny-js/index.js:34]: > Let's start with answering questions.
> 
> > Also why do you need to wrap every presentation string into an array? You have nested arrays as a result.

я вже виправив це

Comment (OlStani) [reply submissions/OlStaniev/tiny-js/index.js:34]: метод  map повертає мені массив строк

Comment (OleksiyRudenko) [reply submissions/OlStaniev/tiny-js/index.js:34]: > я вже виправив це
I can only see what's on PR.

image

Comment (OleksiyRudenko) [reply submissions/OlStaniev/tiny-js/index.js:34]: > метод map повертає мені массив строк
That's how it works. You can glue them with e.g. `.join`

Comment (OleksiyRudenko) [reply submissions/OlStaniev/tiny-js/index.js:34]: I am a peer developer and was asked to add 100 more objects with yet another 5 properties on this project.
Please make my life easier.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
Please make properties separated exactly with `'; '`

**Review by OleksiyRudenko — APPROVED** [review]
@OlStani this works, grats.

I guess your latest change was inspired by other soultions. However, your own initial approach might have been improved with a minor change reducing code repetition. E.g.
```
const inhabitantsInformation = inhabitants.map(({name, gender, legs, hands,saying}) => [name, gender, legs, hands, saying].join('; '))
```

**Comment by OlStani** [comment]
то було занадто просто, тому не прийшло мені в глову)

**Pull request merged**